corehq/messaging/smsbackends/http/models.py

```diff
@@ -163,3 +160,12 @@ def send(self, msg, *args, **kwargs):
         except Exception as e:
             msg = "Error sending message from backend: '{}'\n\n{}".format(self.pk, str(e))
             six.reraise(BackendProcessingException, BackendProcessingException(msg), sys.exc_info()[2])
+
+    @staticmethod
+    def _encode_http_message(text, charset=None):
+        if charset:
+            return text.encode(charset)
+        try:
+            return text.encode("iso-8859-1")
```

Comment:
Do we want is something like
```suggestion
        try:
            return text.encode("iso-8859-1" if charset else charset)
```
that would keep the fallback to utf-8? I'm honestly not sure which is worse (failing hard versus sending what we know will be garbled) though maybe the point here is that this is just a stepping stone to changing _charset_ to utf-8 which obviates the point.

Reply:
Did you mean:
```python
        try:
            return text.encode(charset if charset else "iso-8859-1")
```

> this is just a stepping stone to changing charset to utf-8 which obviates the point

That's what I had in mind.  What I'm hoping to happen is:

1. Deploy this change
1. Swap charset to `utf-8`, test out immediately
1. Swap charset back to `iso-8859-1` if that doesn't work out
1. Drop the first two lines of this method and override `_encode_http_message` only for the `sislog` subclass (so it always encodes as `utf-8`)
1. Deploy that code, and drop `_charset_` from the parameters list in https://www.commcarehq.org/hq/sms/edit_global_gateway/SISLOG/7/

It sounded like they had been previously sending as `iso-8859-1`, so I assumed that was safe, but I'd be happy to keep in the try/except if that feels safer.

Reply:
I should note a couple assumptions:

* `_charset_` is something specific to the sislog gateway
* We wouldn't generally want to modify HQ behavior by these additional params, since they're sent as part of the payload.  We could always do something hacky that strips a magic param, but of course that's best avoided.